### PR TITLE
new setting: usetakefocus

### PR DESCRIPTION
**Pull request by Kreyren**
As referenced on https://github.com/PhoenicisOrg/scripts/pull/939 Unity-based wineapps like **Magic: The Gathering arena** (https://github.com/PhoenicisOrg/scripts/pull/937) and **Minion Masters** (https://github.com/PhoenicisOrg/scripts/issues/957) require registry change of `HKEY_CURRENT_USER\Software\Wine\X11 Driver\UseTakeFocus` to be set on value of `N/Y` accordingly to avoid wineapps from becaming unresponsible to user-input after focus change.

This commit should make it easier for everyone affected to fix the issue easily using winetricks.

Credit:
- vanilline/Constantine for ukranian translate of function
- Community of Learn Klingon for Tlhngan-Hol translate of function

Signed-off-by: Jacob Hrbek <werifgx@gmail.com>

### CODE REVIEW

**Comment by austin987** [comment]
If you can't make the translation, why worry about it? Lots of things are untranslated. And if you are worried about it, why only Ukrainian and not other languages?

**Comment by Kreyren** [comment]
@austin987 
> If you can't make the translation, why worry about it? Lots of things are untranslated. And if you are worried about it, why only Ukrainian and not other languages?

Because every other setting has it and because of perfectionism assuming that we are in FOSS and we can afford making quality code :3 

i was assuming that winetricks dev is russian based on only russian translate present? Was hoping {s,}he can translate it;  else wait i make someone else translate it.

~~didn't know that it's ukranian~~

**Comment by austin987** [comment]
I'm the maintainer, I don't speak Ukrainian or Russian ;). There are active contributors in a few languages that tend to keep things up to date, fwiw.

File: src/winetricks [hunk 1]
```diff
@@ -20337,6 +20337,36 @@ w_metadata windowmanagerdecorated=n settings \
     title_uk="Не дозволяти менеджеру вікон декорувати вікна" \
     title="Prevent the window manager from decorating windows"
 
+#----------------------------------------------------------------
+
+w_metadata usetakefocus=enabled settings \
+    title_uk="Enable UseTakeFocus" \ # Translate needed
+    title="Enable UseTakeFocus"
+w_metadata usetakefocus=disabled settings \
+    title_uk="Disable UseTakeFocus (default)" \ # Translate needed
```
Comment (vanilline) [inline src/winetricks]: Вимкнути фокусування на вікні (за замовчуванням)

File: src/winetricks [hunk 2]
```diff
@@ -20337,6 +20337,36 @@ w_metadata windowmanagerdecorated=n settings \
     title_uk="Не дозволяти менеджеру вікон декорувати вікна" \
     title="Prevent the window manager from decorating windows"
 
+#----------------------------------------------------------------
+
+w_metadata usetakefocus=enabled settings \
+    title_uk="Enable UseTakeFocus" \ # Translate needed
+    title="Enable UseTakeFocus"
```
Comment (vanilline) [inline src/winetricks]: Увімкнути фокусування на вікні

File: src/winetricks [hunk 3]
```diff
@@ -20362,6 +20392,8 @@ w_metadata windowmanagermanaged=n settings \
     title_uk="Не дозволяти менеджеру вікон керування вікнами" \
```
Comment (vanilline) [inline src/winetricks]: Заборонити віконному менеджеру курувати вікнами

File: src/winetricks [hunk 4]
```diff
@@ -20337,6 +20337,36 @@ w_metadata windowmanagerdecorated=n settings \
     title_uk="Не дозволяти менеджеру вікон декорувати вікна" \
```
Comment (vanilline) [inline src/winetricks]: Заборонити віконному менеджеру керувати вікнами

File: src/winetricks [hunk 5]
```diff
@@ -20337,6 +20337,36 @@ w_metadata windowmanagerdecorated=n settings \
...
     title="Prevent the window manager from decorating windows"
 
+#----------------------------------------------------------------
+
+w_metadata usetakefocus=enabled settings \
+    title_uk="Enable UseTakeFocus" \ # Translate needed
```
Comment (vanilline) [inline src/winetricks]: Увімкнути фокусування на вікні

File: src/winetricks [hunk 6]
```diff
@@ -20337,6 +20337,36 @@ w_metadata windowmanagerdecorated=n settings \
...
+    title="Enable UseTakeFocus"
+w_metadata usetakefocus=disabled settings \
+    title_uk="Disable UseTakeFocus (default)" \ # Translate needed
+    title="Disable UseTakeFocus (default)"
```
Comment (vanilline) [inline src/winetricks]: Вимкнути фокусування на вікні (за замовчуванням)

File: src/winetricks [hunk 7]
```diff
@@ -20362,6 +20392,8 @@ w_metadata windowmanagermanaged=n settings \
     title_uk="Не дозволяти менеджеру вікон керування вікнами" \
     title="Prevent the window manager from controlling windows"
```
Comment (vanilline) [inline src/winetricks]: Заборонити віконному менеджеру курувати вікнами

**Comment by Kreyren** [comment]
@vanilline You are a native ukranian speaker correct?

**Comment by vanilline** [comment]
> @vanilline You are a native ukranian speaker correct?

yes, and russian native speaker too

File: src/winetricks (hunk 1 above)
Comment (Kreyren) [reply src/winetricks]: Adapted

File: src/winetricks (hunk 5 above)
Comment (Kreyren) [reply src/winetricks]: Adapted

File: src/winetricks (hunk 6 above)
Comment (Kreyren) [reply src/winetricks]: Adapted

File: src/winetricks (hunk 3 above)
Comment (Kreyren) [reply src/winetricks]: Is irelevant to this Merge request.

File: src/winetricks (hunk 7 above)
Comment (Kreyren) [reply src/winetricks]: Is irelevant to this Merge request.

File: src/winetricks (hunk 4 above)
Comment (Kreyren) [reply src/winetricks]: Is irelevant to this Merge request.

**Comment by Kreyren** [comment]
Contributors for translate credited in original message of MR, Added translate for cz,sk,tlh,uk

**Comment by Kreyren** [comment]
Requesting approval of changes, do not merge yet since tlh might be improved.

**Comment by austin987** [comment]
I don't see a need for Klingon, but easter eggs are fine, I guess :p

**Comment by Kreyren** [comment]
@austin987 Like i can remove it if you don't want that translate :thinking: i was more thinking about actuall usecase since my system is partially in this lang.

**Comment by austin987** [comment]
It's fine. Is this okay to merge then?

**Comment by Kreyren** [comment]
@austin987 yep feel free to merge.

**Comment by Chiitoo** [comment]
As a sidey-note, edits here do not reach people via e-mail...  :]

**Comment by Kreyren** [comment]
@Chiitoo is winetricks beeing monitored via emails?

**Comment by Chiitoo** [comment]
Well, not sure how Austin does it exactly, but when you 'watch' a repository at GitHub, you get notifications, and some only use e-mail notifications, which are not sent when you edit a comment.

So what I would suggest to anyone using GitHub, is not using editing comments for updating things that have any importance since a lot of people won't notice them (I did not see your edits until the new comments came in, and I scrolled up after reading them).

It's not a good system on GitHub's part.

**Comment by Kreyren** [comment]
@Chiitoo is noted

**Comment by austin987** [comment]
Generally I get notifications via mail, then read on the website. edits (and force pushes) do not give notifications; for fixing PRs force pushes are good practice, a comment ensures people know about it.

**Pull request merged**